### ISSUE TITLE
Linked spreadsheet: Write responses to the same worksheet every time

### ISSUE THREAD

**Issue by matthijskooijman** [nextcloud/forms#1971]
I previously posted this as feedback on the PR before it was applied (https://github.com/nextcloud/forms/pull/1758#issuecomment-1896763727), but since it was out of scope for the initial PR, I'm reposting this here.

## Nextcloud:
- Nextcloud-Version: 28.0.1
- Forms-Version: 4.0.0 with PR #1758 applied

## Is your feature request related to a problem? Please describe.
I'm considering to use the forms plugin as an easy way for non-technical users to contribute to a "logbook" spreadsheet. To provide additional info to regular users of the spreadsheet, I would like to add extra worksheets to the spreadsheet, containing extra instructions, or maybe more detailed info.

However, when the spreadsheet has multiple worksheets, results are written to the active worksheet (i.e. the one that was selected when the spreadsheet was saved), even when it was previously saved to another worksheet.

This was slightly unexpected, and makes adding a worksheet with instructions infeasible (unless users always make sure to switch back to the logbook worksheet, but I'm pretty sure that won't happen every time).

## Describe the solution you'd like
Save the response to the same worksheet as the first time.

Maybe some worksheet ID could be saved when creating the spreadsheet and then used to save to the same worksheet every time (the worksheet title would not work so well, since users can change that in the spreadsheet).

**Comment by susnux** [comment]
Could you retry it with forms 4.1? Because for me it works as expected.
We always open the first sheet:

https://github.com/nextcloud/forms/blob/f4c0dc7033cbfb4f6121a2006f452d669d1af222/lib/Service/SubmissionService.php#L332

**Comment by Chartman123** [comment]
No feedback, so closing it for now. Please re-open if issue still persists

**Comment by matthijskooijman** [comment]
I just tried again with 5.1.0, and it indeed seems to use the first sheet. Maybe in my first report I had added a worksheet *before* the form results sheet, and saw it writing results to that one and thought it wrote to the active sheet.

This could still be improved by using the same sheet as before, even when it is no longer the first, but writing to the first sheet is reasonable and usable as well, so also fine as-is now.
